python_modules/dagster/dagster/_core/execution/plan/outputs.py

```diff
@@ -166,8 +166,9 @@ class UnresolvedStepOutputHandle(
         [
             ("unresolved_step_handle", UnresolvedStepHandle),
             ("output_name", str),
-            ("resolved_by_step_key", str),
-            ("resolved_by_output_name", str),
+            ("resolution_source", tuple)
```

Comment:
is this a tuple of resolution sources or a single resolution source represented as a tuple? i thought it was supposed to be able to handle a list of sources but it looks like it's only one in the `resolve` fn

Reply:
lol i was just lazy and used `tuple` since `Any` wasn't imported yet

the end state of this i'm guessing will be `resolution_sources: Sequence[Union[FromPendingDynamicStepOutput,FromUnresolvedStepOutput,FromUnresolvedDynamicStepOutput]]`

we should probably create a shorthand for that union or give a base class to those input sources

Reply:
ok! and then the resolve would iterate through each of the sources and call it's `resolve`?

Reply:
i think thats the gist of it -there may be some cascading related changes